x/ibc/02-client/handler.go

```diff
@@ -0,0 +1,94 @@
+package ics02
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// NewHandler creates a new Handler instance for IBC client
+// transactions
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		ctx = ctx.WithEventManager(sdk.NewEventManager())
+
+		switch msg := msg.(type) {
+		case MsgCreateClient:
+			return handleMsgCreateClient(ctx, k, msg)
+
+		case MsgUpdateClient:
+			return handleMsgUpdateClient(ctx, k, msg)
+
+		case MsgSubmitMisbehaviour:
+			return handleMsgSubmitMisbehaviour(ctx, k, msg)
+
+		default:
+			errMsg := fmt.Sprintf("unrecognized IBC Client message type: %T", msg)
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+func handleMsgCreateClient(ctx sdk.Context, k Keeper, msg MsgCreateClient) sdk.Result {
+	_, err := k.CreateClient(ctx, msg.ClientID, msg.ClientType, msg.ConsensusState)
+	if err != nil {
+		return sdk.ResultFromError(err)
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
```

Comment:
Is there a reason why this requires 2 `sdk.NewEvent` calls? Could we move these into 1 call?

Reply:
this is the standard on the rest of the handlers. I believe is to distinguish msg specific attributes from general ones (i.e sender and category)